File: wagerproof-ios-native/Wagerproof/Features/Analytics/Components/RegressionModelBreakdownSection.swift
import SwiftUI
import WagerproofDesign
import WagerproofModels
import WagerproofStores

/// Day-of-Week & Team Breakdown: segmented bet-type tabs over two ranked
/// tables — by day (Mon..Sun) and by team (sorted by ROI, with logos).
/// Mirrors RN `ModelBreakdownBody` reading `mlb_model_breakdown_accuracy`.
struct RegressionModelBreakdownSection: View {
    let store: MLBModelBreakdownStore
    @State private var tab: String = "full_ml"

    var body: some View {
        if store.loading && store.rows.isEmpty {
            skeleton.transition(.opacity)
        } else if !store.rows.isEmpty {
            let dowRows = store.dowRows(betType: tab)
            let teamRows = store.teamRows(betType: tab)

            VStack(alignment: .leading, spacing: 0) {
                RegressionSegmentedTabs(options: Regression.betTypes, selection: $tab)

                if !dowRows.isEmpty {
                    table(title: "BY DAY OF WEEK", valueLabel: "DAY", rows: dowRows, showLogo: false)
                }
                if !teamRows.isEmpty {
                    table(title: "BY TEAM (SORTED BY ROI)", valueLabel: "TEAM", rows: teamRows, showLogo: true)
                }
                if dowRows.isEmpty && teamRows.isEmpty {
                    Text("No graded picks yet for this bet type.")
                        .font(.system(size: 12)).italic()
                        .foregroundStyle(Color.appTextSecondary)
                        .padding(.top, 12)
                }
            }
        }
        // RN returns nothing when the table has no rows at all.
    }

    private func table(title: String, valueLabel: String, rows: [MLBModelBreakdownRow], showLogo: Bool) -> some View {
        VStack(alignment: .leading, spacing: 4) {
            Text(title)
                .font(.system(size: 10, weight: .bold))
                .tracking(0.5)
                .foregroundStyle(Color.appTextSecondary)
                .padding(.bottom, 2)

            HStack {
                Text(valueLabel).frame(maxWidth: .infinity, alignment: .leading)
                Text("RECORD").frame(width: 66, alignment: .trailing)
                Text("W%").frame(width: 58, alignment: .trailing)
                Text("ROI").frame(width: 68, alignment: .trailing)
            }
            .font(.system(size: 10, weight: .bold))
            .tracking(0.5)
            .foregroundStyle(Color.appTextSecondary)
            .padding(.horizontal, 10)

            ForEach(Array(rows.enumerated()), id: \.offset) { _, r in
                row(r, showLogo: showLogo)
            }
        }
        .padding(.top, 14)
    }

    private func row(_ r: MLBModelBreakdownRow, showLogo: Bool) -> some View {
        HStack {
            HStack(spacing: 6) {
                if showLogo, let urlString = MLBAbbrLogo.url(forAbbr: r.breakdownValue), let url = URL(string: urlString) {
                    AsyncImage(url: url) { image in
                        image.resizable().scaledToFit()
                    } placeholder: {
                        Color.clear
                    }
                    .frame(width: 18, height: 18)
                }
                Text(r.breakdownValue)
                    .font(.system(size: 12, weight: .semibold))
                    .lineLimit(1)
                    .foregroundStyle(Color.appTextPrimary)
            }
            .frame(maxWidth: .infinity, alignment: .leading)

            Text("\(r.wins)-\(r.losses)" + (r.pushes > 0 ? "-\(r.pushes)" : ""))
                .font(.system(size: 12))
                .monospacedDigit()
                .frame(width: 66, alignment: .trailing)
                .foregroundStyle(Color.appTextSecondary)
            Text(Regression.trimmed(r.winPct) + "%")
                .font(.system(size: 12, weight: .bold))
                .monospacedDigit()
                .frame(width: 58, alignment: .trailing)
                .foregroundStyle(Regression.winPctColor(r.winPct))
            Text(Regression.signedTrimmedPct(r.roiPct))
                .font(.system(size: 12, weight: .semibold))
                .monospacedDigit()
                .frame(width: 68, alignment: .trailing)
                .foregroundStyle(Regression.roiColor(r.roiPct))
        }
        .padding(.horizontal, 10)
        .padding(.vertical, 8)
        .background(Color.appSurfaceMuted.opacity(0.3), in: RoundedRectangle(cornerRadius: 8))
    }

    private var skeleton: some View {
        VStack(spacing: 10) {
            SkeletonBlock(height: 34, cornerRadius: 10)
                .shimmering()
            VStack(spacing: 4) {
                ForEach(0..<5, id: \.self) { _ in
                    HStack {
                        SkeletonBlock(width: 90, height: 12).frame(maxWidth: .infinity, alignment: .leading)
                        SkeletonBlock(width: 44, height: 12).frame(width: 66, alignment: .trailing)
                        SkeletonBlock(width: 34, height: 12).frame(width: 58, alignment: .trailing)
                        SkeletonBlock(width: 40, height: 12).frame(width: 68, alignment: .trailing)
                    }
                    .padding(.horizontal, 10)
                    .padding(.vertical, 8)
                    .shimmering()
                    .background(Color.appSurfaceMuted.opacity(0.3), in: RoundedRectangle(cornerRadius: 8))
                }
            }
        }
    }
}

File: wagerproof-android-native/app/src/main/java/com/wagerproof/app/features/analytics/RegressionModelBreakdownSection.kt
package com.wagerproof.app.features.analytics

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.wagerproof.app.features.shared.RemoteImage
import com.wagerproof.core.design.components.SkeletonBlock
import com.wagerproof.core.design.components.shimmering
import com.wagerproof.core.design.tokens.AppColors
import com.wagerproof.core.models.MLBAbbrLogo
import com.wagerproof.core.models.MLBModelBreakdownRow
import com.wagerproof.core.stores.MLBModelBreakdownStore

/**
 * Day-of-Week & Team Breakdown: segmented bet-type tabs over two ranked tables
 * — by day (Mon..Sun) and by team (sorted by ROI, with logos). Port of iOS
 * `RegressionModelBreakdownSection` reading `mlb_model_breakdown_accuracy`.
 */
@Composable
fun RegressionModelBreakdownSection(
    store: MLBModelBreakdownStore,
    modifier: Modifier = Modifier,
) {
    var tab by remember { mutableStateOf("full_ml") }

    Column(modifier.fillMaxWidth()) {
        when {
            store.loading && store.rows.isEmpty() -> Skeleton()
            store.rows.isNotEmpty() -> {
                val dowRows = store.dowRows(tab)
                val teamRows = store.teamRows(tab)
                Column(Modifier.fillMaxWidth()) {
                    RegressionSegmentedTabs(Regression.betTypes, tab, { tab = it })
                    if (dowRows.isNotEmpty()) {
                        Table("BY DAY OF WEEK", "DAY", dowRows, showLogo = false)
                    }
                    if (teamRows.isNotEmpty()) {
                        Table("BY TEAM (SORTED BY ROI)", "TEAM", teamRows, showLogo = true)
                    }
                    if (dowRows.isEmpty() && teamRows.isEmpty()) {
                        Text(
                            text = "No graded picks yet for this bet type.",
                            fontSize = 12.sp,
                            fontStyle = FontStyle.Italic,
                            color = AppColors.appTextSecondary,
                            modifier = Modifier.padding(top = 12.dp),
                        )
                    }
                }
            }
            // RN returns nothing when the table has no rows at all.
        }
    }
}

@Composable
private fun Table(
    title: String,
    valueLabel: String,
    rows: List<MLBModelBreakdownRow>,
    showLogo: Boolean,
) {
    Column(Modifier.fillMaxWidth().padding(top = 14.dp), verticalArrangement = Arrangement.spacedBy(4.dp)) {
        Text(
            text = title,
            fontSize = 10.sp,
            fontWeight = FontWeight.Bold,
            letterSpacing = 0.5.sp,
            color = AppColors.appTextSecondary,
            modifier = Modifier.padding(bottom = 2.dp),
        )
        Row(Modifier.fillMaxWidth().padding(horizontal = 10.dp)) {
            HeaderCell(valueLabel, Modifier.weight(1f), TextAlign.Start)
            HeaderCell("RECORD", Modifier.width(66.dp), TextAlign.End)
            HeaderCell("W%", Modifier.width(58.dp), TextAlign.End)
            HeaderCell("ROI", Modifier.width(68.dp), TextAlign.End)
        }
        rows.forEach { r -> BreakdownRow(r, showLogo) }
    }
}

@Composable
private fun BreakdownRow(r: MLBModelBreakdownRow, showLogo: Boolean) {
    val record = "${r.wins}-${r.losses}" + if (r.pushes > 0) "-${r.pushes}" else ""
    Row(
        Modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(8.dp))
            .background(AppColors.appSurfaceMuted.copy(alpha = 0.3f))
            .padding(horizontal = 10.dp, vertical = 8.dp),
        verticalAlignment = Alignment.CenterVertically,
    ) {
        Row(
            Modifier.weight(1f),
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.spacedBy(6.dp),
        ) {
            if (showLogo) {
                val url = MLBAbbrLogo.url(r.breakdownValue)
                if (url != null) {
                    RemoteImage(
                        url = url,
                        contentDescription = r.breakdownValue,
                        modifier = Modifier.size(18.dp),
                    )
                }
            }
            Text(
                text = r.breakdownValue,
                fontSize = 12.sp,
                fontWeight = FontWeight.SemiBold,
                color = AppColors.appTextPrimary,
                maxLines = 1,
                overflow = TextOverflow.Ellipsis,
            )
        }
        Text(
            text = record,
            fontSize = 12.sp,
            fontFamily = FontFamily.Monospace,
            textAlign = TextAlign.End,
            color = AppColors.appTextSecondary,
            modifier = Modifier.width(66.dp),
        )
        Text(
            text = Regression.trimmed(r.winPct) + "%",
            fontSize = 12.sp,
            fontWeight = FontWeight.Bold,
            fontFamily = FontFamily.Monospace,
            textAlign = TextAlign.End,
            color = Regression.winPctColor(r.winPct),
            modifier = Modifier.width(58.dp),
        )
        Text(
            text = Regression.signedTrimmedPct(r.roiPct),
            fontSize = 12.sp,
            fontWeight = FontWeight.SemiBold,
            fontFamily = FontFamily.Monospace,
            textAlign = TextAlign.End,
            color = Regression.roiColor(r.roiPct),
            modifier = Modifier.width(68.dp),
        )
    }
}

@Composable
private fun HeaderCell(text: String, modifier: Modifier, align: TextAlign) {
    Text(
        text = text,
        fontSize = 10.sp,
        fontWeight = FontWeight.Bold,
        letterSpacing = 0.5.sp,
        color = AppColors.appTextSecondary,
        textAlign = align,
        modifier = modifier,
    )
}

@Composable
private fun Skeleton() {
    Column(verticalArrangement = Arrangement.spacedBy(10.dp)) {
        SkeletonBlock(height = 34.dp, cornerRadius = 10.dp, modifier = Modifier.shimmering())
        Column(verticalArrangement = Arrangement.spacedBy(4.dp)) {
            repeat(5) {
                Row(
                    Modifier
                        .fillMaxWidth()
                        .clip(RoundedCornerShape(8.dp))
                        .background(AppColors.appSurfaceMuted.copy(alpha = 0.3f))
                        .padding(horizontal = 10.dp, vertical = 8.dp)
                        .shimmering(),
                ) {
                    SkeletonBlock(width = 90.dp, height = 12.dp, modifier = Modifier.weight(1f))
                    Box(Modifier.width(6.dp))
                    SkeletonBlock(width = 44.dp, height = 12.dp)
                }
            }
        }
    }
}
